### PR TITLE
Replace `mk_const` with `Const::new_x` methods

### PR DESCRIPTION
**Pull request by BoxyUwU**
Part of rust-lang/compiler-team#616. Instead of just havign `Const::new(` and nothing else I did it like this since this is more like how the `mk_x` works for `Ty`, and also another PR of mine will require changing from `Const::new(` to `Const::new_x(` anyway.

r? @oli-bok

### CODE REVIEW

File: compiler/rustc_symbol_mangling/src/v0.rs [hunk 1]
```diff
@@ -651,7 +651,7 @@ impl<'tcx> Printer<'tcx> for &mut SymbolMangler<'tcx> {
                             .builtin_deref(true)
                             .expect("tried to dereference on non-ptr type")
                             .ty;
-                        let dereferenced_const = self.tcx.mk_const(ct.kind(), pointee_ty);
+                        let dereferenced_const = self.tcx.mk_ct_from_kind(ct.kind(), pointee_ty);
```
Comment (lcnr) [inline compiler/rustc_symbol_mangling/src/v0.rs:654]:
```suggestion
                        // FIXME: add an assert that we only do this for valtrees.
                        let dereferenced_const = self.tcx.mk_ct_from_kind(ct.kind(), pointee_ty);
```

**Review by lcnr — COMMENTED** [review]
r=me after nit

**Comment by jyn514** [comment]
lol i think bors didn't like that my comment was so long

@bors retry

**Pull request merged**